tidy3d/components/data/monitor_data.py

```diff
@@ -133,7 +133,136 @@ def symmetry_expanded_copy(self) -> AbstractFieldData:
         return self.copy(update=update_dict)
 
 
-class FieldData(FieldDataset, AbstractFieldData):
+class ElectromagneticFieldData(AbstractFieldData, ElectromagneticFieldDataset, ABC):
+    """Collection of electromagnetic fields."""
+
+    @property
+    def _tangential_dims(self) -> List[str]:
+        """For a 2D monitor data, return the names of the tangential dimensions. Raise if cannot
+        confirm that the associated monitor is 2D."""
+        zero_dims = np.where(np.array(self.monitor.size) == 0)[0]
```

Comment:
doesn't have to be here, but I feel like this kind of thing comes up so often that we might want a `Box.num_nonzero_dims` property or something.

Reply:
and maybe a `Box.normal_axis` property also. (Errors if `Box.num_nonzero_dims != 1`)

Reply:
So should I add this to this PR?